### ISSUE TITLE
No Tests Run

### ISSUE THREAD

**Issue by MarcusFulbright** [apiaryio/dredd#126]
I'm trying to use Dredd for the first time. I have dredd v0.3.14 installed and running on OSX. Blue print file is a copy and paste from this example: https://gist.github.com/netmilk/6885268

Below is a copy and paste from my terminal:

$ dredd blueprint.apib http://localhost:3000/
info: Beginning Dredd testing...

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: connect ECONNREFUSED
  at errnoException (net.js:905:11)
  at Object.afterConnect [as oncomplete] (net.js:896:19)

I am running a php Server on :3000, php -S localhost:3000. I can view it in my browser. Currently it just returns a 404 as it does not point to anything. I'm not sure what I'm doing wrong. Any help or input would be appreciated.

**EDIT**
omitting trailing slash seems to change behavior:
dredd blueprint.apib localhost:3000
info: Beginning Dredd testing...

/usr/local/lib/node_modules/dredd/lib/transaction-runner.js:67
      fullPath = '/' + [parsedUrl['path'].replace(/^\/|\/$/g, ""), request['ur
                                          ^
TypeError: Cannot call method 'replace' of null
  at TransactionRunner.configureTransaction (/usr/local/lib/node_modules/dredd/lib/transaction-runner.js:67:43)
  at /usr/local/lib/node_modules/dredd/lib/transaction-runner.js:3:59
  at /usr/local/lib/node_modules/dredd/node_modules/async/lib/async.js:249:17
  at iterate (/usr/local/lib/node_modules/dredd/node_modules/async/lib/async.js:149:13)
  at async.eachSeries (/usr/local/lib/node_modules/dredd/node_modules/async/lib/async.js:165:9)
  at _asyncMap (/usr/local/lib/node_modules/dredd/node_modules/async/lib/async.js:248:13)
  at Object.mapSeries (/usr/local/lib/node_modules/dredd/node_modules/async/lib/async.js:231:23)
  at TransactionRunner.run (/usr/local/lib/node_modules/dredd/lib/transaction-runner.js:50:11)
  at /usr/local/lib/node_modules/dredd/lib/dredd.js:86:29

**Comment by MarcusFulbright** [comment]
This issue can be closed. The problem was with how the php web server behaves. For future reference, I got the php server to work by doing this:

`php -S 0.0.0.0:8000`

Not entirely sure why, but hopefully this helps others.

**Comment by ramsey** [comment]
Sweet, @MarcusFulbright. That got it working for me! :-)

**Comment by abtris** [comment]
:+1:

**Comment by netmilk** [comment]
@MarcusFulbright I'm sorry I totally didn't notice your EDIT in the original issue. It's very important, but I can't replicate it. Is this still valid? What version of Dredd and Node.js do you use?

**Comment by MarcusFulbright** [comment]
As originally listed, I had dredd v0.3.14, not sure about the node version. The machine I was using is no longer in my possession so I cannot check. Furthermore, the office has migrated a Windows environment. Since then I haven't been able to install dredd. One of the dependencies, either poltergeist or protagonist, caused me lots of grief. I checked out their repos for troubleshooting help to no avail, though others seem to report installing them just fine.

Unfortunately, I can't do anything to try and replicate the issue at this time. Thanks for taking the time to re-open the issue. Hopefully you can get to the bottom of it. Wish I could help more.

**Comment by netmilk** [comment]
@MarcusFulbright thanks a lot for finding time to provide me this feedback! It's super helpful even if you don't have access to the machine anymore. Your stress on missing of support for Windows is what's very important as well. I created an issue regarding that #204.

Do I assume correctly that's the reason why you can't update Dredd to the latest which is 0.5.4 at this moment?

**Comment by MarcusFulbright** [comment]
@netmilk correct. Again, the installation woes I'm having on windows seem to be due to a dependency, not dredd itself.

**Comment by JefStat** [comment]
The issue title is correct for me, but the description is not the same as mine. 

**TLDR:** Dredd sometimes doesn't run any tests not even validating the blueprint. It just exits without any messaging at all. I cannot reproduce it well, I think it can be reproduced by running dredd several times in a row, such as while testing the dredd configuration settings.

Current setup 
I have a batch file that runs a powershell file that spin up the api host and then runs dredd. The batch file is ran in Teamcity and the Dredd junit report is parsed by Teamcity.

### Issue 1:
 - Dredd, well apiary doesn't work with windows line endings.

This .gitattributes file works around this issue.
````
# apiary.apib requires lf line endings
apiary.apib text eol=lf
````

### Issue 2: 
 - Dredd xaml configuration doesn't start the api service in the background

This powershell file fixes an issue when using Dredd xaml for the startup. 
```
node Elasticsearch/Setup.js
Start-Process -FilePath "api.exe" -PassThru
# wait for api to startup 
Start-Sleep -s 3
node dredd.js
Stop-Process -Name "api.exe"
node Elasticsearch/Teardown.js
```

### Issue 3:
- Node_gyp is slow and half the time it fails for one reason or another. ie missing MSBuild environment variables. So it's often a pain to use `npm install dredd`

To workaround the tooling issues installing Dredd from npm I committed the node_modules to the repo. About 140mb tax on the repo size.

**Comment by netmilk** [comment]
@MarcusFulbright Created new issue for crashing Dredd mentioned in **EDIT** here: #260

**Comment by netmilk** [comment]
@JefStat Thank you very much for the feedback! There is a generic issue regarding Windows support here #204. Let's continue the discussion there.
